hash.go

```diff
@@ -6,6 +6,7 @@ import "C"
 import (
 	"crypto"
 	"errors"
+	"fmt"
```

Comment:
Can't use `fmt` in this package, as it is not allowed as a dependency of `crypto` packages.

Same for the CNG and Darwin backends.